### ISSUE TITLE
Opposite preditions

### ISSUE THREAD

**Issue by nkadochn** [amaiya/ktrain#277]
I am using ktrain 0.25.0.  My model performance (text classification using Bi-GRU) is excellent (about 95% accuracy), however my predictions are always opposite.  How can I get the predictions to align with original values?

![image](https://user-images.githubusercontent.com/56486956/99984273-7822cb80-2d72-11eb-83d6-e9c4530e4041.png)

**Comment by amaiya** [comment]
Which `texts_from*` function did you use and what does the original training data look like (both text and labels)?

**Comment by nkadochn** [comment]
@amaiya 

When I use "texts_from_df" (see attached ktrain_Classifier_biGRU_yelp_pandas.ipynb) it works fine.  But when I use "texts_from_array"  (see attached ktrain_Classifier_biGRU_yelp_series.ipynb) my predictions are opposite.


[ktrain_Classifier_biGRU_yelp_pandas.zip](https://github.com/amaiya/ktrain/files/5584454/ktrain_Classifier_biGRU_yelp_pandas.zip)
[ktrain_Classifier_biGRU_yelp_series.zip](https://github.com/amaiya/ktrain/files/5584455/ktrain_Classifier_biGRU_yelp_series.zip)

**Comment by amaiya** [comment]
In comparing the notebooks, I think this may be a bug in your preprocessing code when one-hot-encoding the labels.  The notebooks don't show the values of the arrays just prior to invoking `texts_from_array`, but you can verify the bug yourself.

When you use `LabelEncoder`, I believe the integers are assigned to labels in alphabetical order (Negative=0, Positive=1).  However, when you do something like `target_names = np.array(df.target.unique()).tolist()`, the targets are not necessarily in alphabetical order.  The problem is that you're supplying `target_names` to `texts_from_array`.  The `texts_from_df` invocation doesn't suffer from this problem because you're not using `target_names` in the same way.

To avoid such problems, you might try removing this preprocessing that you're doing yourself because **ktrain** can now handle this itself.  For instance, note that both `texts_from_df` and `texts_from_array` no longer require that you one-hot-encode the labels yourself, as shown in [this notebook](https://nbviewer.jupyter.org/github/amaiya/ktrain/blob/master/examples/text/ArabicHotelReviews-AraBERT.ipynb).  For instance, see the doc string for `texts_from_array`:
```python
def texts_from_array(x_train, y_train, x_test=None, y_test=None,
                   class_names = [],
                   max_features=MAX_FEATURES, maxlen=MAXLEN,
                   val_pct=0.1, ngram_range=1, preprocess_mode='standard',
                   lang=None, # auto-detected
                   random_state=None,
                   verbose=1):
    """
    Loads and preprocesses text data from arrays.
    texts_from_array can handle data for both text classification
    and text regression.  If class_names is empty, a regression task is assumed.
    Args:
        x_train(list): list of training texts 
        y_train(list): labels in one of the following forms:
        x_test(list): list of training texts 
        y_test(list): labels in one of the following forms:
                       1. list of integers representing classes (class_names is required)
                       2. list of strings representing classes (class_names is not needed and ignored.)
                       3. a one or multi hot encoded array representing classes (class_names is required)
                       4. numerical values for text regresssion (class_names should be left empty)
        class_names (list): list of strings representing class labels
                            shape should be (num_examples,1) or (num_examples,)
```

I'll close this issue, but feel free to reply to this thread, if there are further questions or problems.

**Comment by nkadochn** [comment]
Thanks a lot for your guidance @amaiya 

It is so great to see the number of enhancements you put into the library, I was upgrading the code from version 0.3.0 and you made it a lot easier to use.

Hopefully TensorFlow 2.4.0 will be released in the next few week, so we can get original "learner.lr_find" functionality back
